### PR TITLE
[SAIC-456] Add DB config options for identity (keystone) service in CF

### PR DESCRIPTION
**Pull request by mirrorcoder**

### CODE REVIEW

**Comment by realsystem** [comment]
+1

File: cfglib.py [hunk 1]
```diff
@@ -174,6 +174,16 @@
...
+    cfg.StrOpt('db_user', default=None,
+               help='user for mysql'),
+    cfg.StrOpt('db_password', default=None,
+               help='password for mysql'),
+    cfg.StrOpt('db_host', default=None,
+               help='host of mysql'),
+    cfg.IntOpt('db_port', default=None,
+               help='port for mysql connection'),
+    cfg.StrOpt('db_connection', default=None,
+               help='driver for connection'),
```
Comment (roman-verchikov) [inline cfglib.py]: I might be missing something, but do we really need DB creds for rabbit?.. My understanding is that for rabbit we only need rabbit-specific creds, not mysql.

Comment (miarmak) [inline cfglib.py]: No, we do not need DB creds for RabbitMQ. For Rabbit we only need Rabbit-specific creds, not mysql. And also we use it only in our checks. So i guess we can remove it form this patch.

File: cfglib.py [hunk 2]
```diff
@@ -273,6 +283,16 @@
                             title='Config service for identity')
 
 src_identity_opts = [
+    cfg.StrOpt('db_user', default=None,
```
Comment (miarmak) [inline cfglib.py]: I guess we can provide some default values here. F.e. 
```
db_user = keystone
db_password = keystone
db_name = keystone
db_connection = mysql+mysqlconnector
db_port = 3306
```

File: cfglib.py [hunk 3]
```diff
@@ -273,6 +283,16 @@
...
+               help='password for mysql'),
+    cfg.StrOpt('db_host', default=None,
+               help='host of mysql'),
+    cfg.IntOpt('db_port', default=None,
+               help='port for mysql connection'),
+    cfg.StrOpt('db_connection', default=None,
+               help='driver for connection'),
     cfg.StrOpt('service', default='keystone',
                help='name service for keystone'),
     cfg.StrOpt('database_name', default='',
```
Comment (miarmak) [inline cfglib.py]: It seems as this option is obsolete and should be renamed to `db_name`.

File: cfglib.py [hunk 4]
```diff
@@ -305,6 +325,16 @@
                               title='Config service for object storage')
 
 src_objstorage_opts = [
```
Comment (miarmak) [inline cfglib.py]: In commit message you've specified only about identity (keystone) DB config options adding. Remove it from this patch OR add info in the commit message about it please.

File: cfglib.py [hunk 5]
```diff
@@ -305,6 +325,16 @@
...
+               help='password for mysql'),
+    cfg.StrOpt('db_host', default=None,
+               help='host of mysql'),
+    cfg.IntOpt('db_port', default=None,
+               help='port for mysql connection'),
+    cfg.StrOpt('db_connection', default=None,
+               help='driver for connection'),
     cfg.StrOpt('service', default='swift',
                help='service name for object storage'),
     cfg.StrOpt('database_name', default='',
```
Comment (miarmak) [inline cfglib.py]: The same issue as above about `db_name`.

File: cfglib.py [hunk 6]
```diff
@@ -331,6 +361,16 @@
                           title='Config RabbitMQ for source cloud')
 
 dst_rabbit_opts = [
```
Comment (miarmak) [inline cfglib.py]: And everything as above for DST.

File: cfglib.py (hunk 2 above)
Comment (mirrorcoder) [inline cfglib.py]: Default params must be None and cloudferry will use _mysql sections creds

Comment (miarmak) [inline cfglib.py]: OK

File: cfglib.py (hunk 1 above)
Comment (roman-verchikov) [inline cfglib.py]: @mirrorcoder please remove config options

**Comment by roman-verchikov** [comment]
+1, can be merged.

**Pull request merged**